engine-tests/src/tests/xcc.rs

```diff
@@ -134,9 +134,14 @@ fn test_xcc_eth_gas_cost() {
         costs::CROSS_CONTRACT_CALL_BYTE
     );
 
-    // As a sanity check, confirm that the total EVM gas spent aligns with expectations
-    let total_gas1 = y1 + baseline.all_gas();
-    let total_gas2 = y2 + baseline.all_gas();
+    // As a sanity check, confirm that the total EVM gas spent aligns with expectations.
+    // The additional gas added is the amount attached to the XCC call (this is "used", but not
+    // "burnt").
+    let total_gas1 = y1 + baseline.all_gas() + costs::ROUTER_EXEC_BASE.as_u64();
+    let total_gas2 = y2
+        + baseline.all_gas()
+        + costs::ROUTER_EXEC_BASE.as_u64()
+        + costs::ROUTER_EXEC_PER_CALLBACK.as_u64();
     assert!(
```

Comment:
Just noting now that assertions that could lead to failure in the future shouldn't be a thing, only errors.